express-post-json/index.js

```diff
@@ -0,0 +1,30 @@
+var express = require('express');
+var app = express();
+express.json();
+app.use(express.json());
+
+var nextId = 1;
+var grades = {};
+
+app.get('/api/grades', (req, res) => {
+  var gradesArray = [];
+  for (var key in grades) {
+    gradesArray.push(grades[key]);
+  }
+  res.json(gradesArray);
+});
+
+app.post('/api/grades', (req, res) => {
+  res.setHeader('content-type', 'application/json', 'charset Utf-8');
+  var gradeObject = req.body;
+  gradeObject.id = nextId;
+  grades[nextId] = gradeObject;
+  nextId++;
+  res.json(grades);
+  res.status(201);
```

Comment:
Be sure to set the status code before sending the body. Otherwise, the custom 201 gets ignored.

Reply:
I fixed it